### ISSUE TITLE
Heartbeat 8.5.0: state.duration_ms is improperly mapped as type date instead of long

### ISSUE THREAD

**Issue by untergeek** [elastic/beats#33645]
Not sure if this is affecting any other customers or if it's been fixed in some other way (I didn't see it in any other open or closed issue).

### The problem

The new index template pushed in 8.5.0 is mapping `state.duration_ms` as a `date` type instead of a `float` (since the raw stat appears to have a decimal).

This is confirmed by viewing the mapping in the Index Mapping area of Stack Management in Kibana (will provide screenshot if desired):
```
"state": {
        "properties": {
          "checks": {
            "type": "long"
          },
          "down": {
            "type": "long"
          },
          "duration_ms": {
            "type": "date"
          },
```

This shows up in the heartbeat logs (which is how I found it):
```
Cache:publisher.EventCache{m:mapstr.M(nil)}} (status=400): {\"type\":\"mapper_parsing_exception\",\"reason\":\"failed to parse field [state.duration_ms] of type [date] in document with id 'PpKwYoQB5Yfu5HjNxzT9'. Preview of field's value: '1985243.0'\",\"caused_by\":{\"type\":\"illegal_argument_exception\",\"reason\":\"failed to parse date field [1.985243e+06] with format [strict_date_optional_time||epoch_millis]\",\"caused_by\":{\"type\":\"date_time_parse_exception\",\"reason\":\"date_time_parse_exception: Failed to parse with all enclosed parsers\"}}}, dropping event!","service.name":"heartbeat","ecs.version":"1.6.0"}
```
This effectively fills up the buffer in heartbeat, and I just stop receiving _any_ data from heartbeat. My dashboard is empty after the TLS certificate checks happen, and this `state.duration_ms` field appears there.

I verified the fix by deleting all of my indices/data_streams and templates and letting it all get re-created. I've confirmed that my manual template modification to set `state.duration_ms` to `float` and letting heartbeat re-create the data stream does create the data stream with the proper mapping.

**Comment by Kilhog** [comment]
Same error on my instance, but I can't rewrite the mapping

**Comment by untergeek** [comment]
@Kilhog you can't rewrite an existing mapping, but you can change the template (which is what I did). If you do that, you can delete the data stream and then let it get re-created with the updated mapping from the template.

**Comment by untergeek** [comment]
Evidently this has already been addressed and the fix is in 8.5.1, released today.

https://discuss.elastic.co/t/heartbeat-8-5-0-mapper-error-workaround/318132
